src/service/search/datafusion/table_provider/helpers.rs

```diff
@@ -157,6 +157,33 @@ pub fn generate_access_plan(file: &PartitionedFile) -> Option<Arc<ParquetAccessP
         return None;
     };
     let row_group_count = num_rows.div_ceil(PARQUET_MAX_ROW_GROUP_SIZE);
+
+    // Determine sampling mode based on BitVec size:
+    // - If BitVec size == row_group_count: row-group-level sampling (enterprise feature)
+    // - If BitVec size == num_rows: row-level sampling (original behavior)
+    let is_row_group_sampling = row_ids.len() == row_group_count;
+
+    #[cfg(feature = "enterprise")]
+    if is_row_group_sampling {
+        return Some(
+            o2_enterprise::enterprise::search::sampling::execution::generate_row_group_access_plan(
+                &row_ids,
+                row_group_count,
+                file.path().as_ref(),
+            ),
+        );
+    }
+
+    #[cfg(not(feature = "enterprise"))]
+    if is_row_group_sampling {
+        log::warn!(
+            "[SAMPLING] Row-group-level sampling is an enterprise feature. File will not be scanned: {:?}",
+            file.path().as_ref()
+        );
+        return None;
```

Comment:
i think just skip this feature run the bellow logic is fine. if we return None, it will corrupt the scan logic.